pkg/autogen/autogen.go

```diff
@@ -31,82 +73,44 @@ const (
 //          - mutate.Patches/mutate.PatchesJSON6902/validate.deny/generate rule is defined
 // - otherwise it returns all pod controllers
 func CanAutoGen(spec *kyverno.Spec, log logr.Logger) (applyAutoGen bool, controllers string) {
-	var needAutogen bool
-	rules := spec.Rules
-	for _, rule := range rules {
-		match := rule.MatchResources
-		exclude := rule.ExcludeResources
-
-		if match.ResourceDescription.Name != "" || match.ResourceDescription.Selector != nil || match.ResourceDescription.Annotations != nil ||
-			exclude.ResourceDescription.Name != "" || exclude.ResourceDescription.Selector != nil || exclude.ResourceDescription.Annotations != nil {
-			log.V(3).Info("skip generating rule on pod controllers: Name / Selector in resource description may not be applicable.", "rule", rule.Name)
-			return false, ""
+	needed := false
+	for _, rule := range spec.Rules {
+		if rule.Mutation.PatchesJSON6902 != "" || rule.HasGenerate() {
```

Comment:
Can we also check if the rule is `imageVerify` to return early?

Reply:
@realshuting I think it was not there before right ?

Reply:
Looks like it's missing when we added `imageVerify` type of rule.

Reply:
If it was not present before the refactor I would rather log it into an issue and address it in another PR.